### DISCUSSION TITLE
MDXRenderer is undefined (gatsby-plugin-mdx)

### DISCUSSION THREAD

**Discussion by clarkedb** [gatsbyjs/gatsby#37099] [Help]
I'm upgrading to the latest version (v5) of `gatsby-plugin-mdx` from v3.

Previously I used `MDXRenderer` to render the mdx body.

I import it from the plugin: `import { MDXRenderer } from "gatsby-plugin-mdx"`, but I run into issues because `MDXRenderer` is `undefined`.

I know that [the migration guide for v3 to v4](https://www.gatsbyjs.com/plugins/gatsby-plugin-mdx/#updating-page-templates) mentions that `MDXRenderer` is no longer needed for page templates, but that example does not match the pattern of how any of my site was working in V3.

Is `MDXRenderer` supposed to be `undefined`? Is there no longer a way to render the body of an mdx page returned from a graphql query?

**Comment by NazAnton** [comment]
> I'm upgrading to the latest version (v5) of `gatsby-plugin-mdx` from v3.
> 
> Previously I used `MDXRenderer` to render the mdx body.
> 
> I import it from the plugin: `import { MDXRenderer } from "gatsby-plugin-mdx"`, but I run into issues because `MDXRenderer` is `undefined`.
> 
> I know that [the migration guide for v3 to v4](https://www.gatsbyjs.com/plugins/gatsby-plugin-mdx/#updating-page-templates) mentions that `MDXRenderer` is no longer needed for page templates, but that example does not match the pattern of how any of my site was working in V3.
> 
> Is `MDXRenderer` supposed to be `undefined`? Is there no longer a way to render the body of an mdx page returned from a graphql query?

Please look at [this thread](https://github.com/gatsbyjs/gatsby/discussions/34714). That should help. 

But to directly answer your question, yes, `gatsby-plugin-mdx` v4+ no longer has an MDXRenderer, and the `body` doesn't contain a compiled function like it was in v3. Breaking change.
